lib/tree.rb

```diff
@@ -2,65 +2,150 @@ class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
 
-   def initialize(key, val)
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
+
   def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity:o(log n )
+  # Space Complexity: o(n)
   def add(key, value)
```

Comment:
The space/time complexity is O(log n) if the tree is balanced and O(n) if it's not.
The method however works